executor/split.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"context"
+
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/table/tables"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+// SplitIndexRegionExec represents a split index regions executor.
+type SplitIndexRegionExec struct {
+	baseExecutor
+
+	table      table.Table
+	indexInfo  *model.IndexInfo
+	valueLists [][]types.Datum
+}
+
+type splitableStore interface {
+	SplitRegionAndScatter(splitKey kv.Key) (uint64, error)
+	WaitScatterRegionFinish(regionID uint64) error
+}
+
+// Next implements the Executor Next interface.
+func (e *SplitIndexRegionExec) Next(ctx context.Context, _ *chunk.RecordBatch) error {
+	store := e.ctx.GetStore()
+	s, ok := store.(splitableStore)
+	if !ok {
+		return nil
+	}
+	regionIDs := make([]uint64, 0, len(e.valueLists))
+	index := tables.NewIndex(e.table.Meta().ID, e.table.Meta(), e.indexInfo)
+	for _, values := range e.valueLists {
+		idxKey, _, err := index.GenIndexKey(e.ctx.GetSessionVars().StmtCtx, values, 1, nil)
```

Comment:
`1` means min handle id?

Reply:
yes, Maybe `0` is better?

Reply:
you can use a constant, better than hard code!

Reply:
@winkyao
I think `1` isn't always the min handle. If a table has a PKIsHandle column, then the handle may be a negative value.
So I think we need to use `math.MinInt64`.

Reply:
@zimulala The key is encoding in binary format, so negative value is always greater than a positive value. @nolouch Right?

Reply:
Here is a test:
0                        encode to byte is: [128 0 0 0 0 0 0 0]
math.MinInt64 encode to byte is:  [128 0 0 0 0 0 0 0]

so,  math.MinInt64 is better.